sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample08_ConsumeEvents.cs

```diff
@@ -77,40 +77,49 @@ await using (var inspectionClient = new EventHubProducerClient(connectionString,
-                //
-                // Because the maximum wait time is specified as zero, this call will return immediately and will not
-                // have consumed any events.
-
-                await consumerClient.ReceiveAsync(1, TimeSpan.Zero);
+                PartitionEvent receivedEvent;
 
-                // Now that the consumer is watching the partition, let's publish the event that we would like to
-                // receive.
-
-                await producerClient.SendAsync(new EventData(Encoding.UTF8.GetBytes("Hello, Event Hubs!")));
-                Console.WriteLine("The event batch has been published.");
+                Stopwatch watch = Stopwatch.StartNew();
+                bool wereEventsPublished = false;
 
+                // Because our consumer is reading from the latest position, it won't see events that have previously
+                // been published.  Before we can publish the events and have them observed, we will need to ask the consumer
+                // to perform an operation, because it opens its connection only when it needs to.
+                //
+                // We'll begin to iterate on the partition using a small wait time, so that control will return to our loop even when
+                // no event is available.  For the first call, we'll publish so that we can receive them.
+                //
                 // Because publishing and receiving events is asynchronous, the events that we published may not
-                // be immediately available for our consumer to see.
+                // be immediately available for our consumer to see, so we'll have to guard against an empty event being sent as
+                // punctuation if our actual event is not available within the waiting time period.
                 //
                 // We will iterate over the available events in the partition, which should be just the event that we published.  Because
                 // we're expecting only the one event, we will exit the loop when we receive it.  To be sure that we do not block forever
-                // waiting on an event that is not published, we will specify a fairly long time to wait and then cancel waiting.
+                // waiting on an event that is not published, we will request cancellation after a fairly long interval.
 
                 CancellationTokenSource cancellationSource = new CancellationTokenSource();
                 cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
 
-                EventData receivedEvent = null;
-                Stopwatch watch = Stopwatch.StartNew();
-
-                await foreach (EventData currentEvent in consumerClient.SubscribeToEvents(cancellationSource.Token))
+                await foreach (PartitionEvent currentEvent in consumerClient.ReadEventsFromPartitionAsync(firstPartition, EventPosition.Latest, TimeSpan.FromMilliseconds(150), cancellationSource.Token))
```

Comment:
We're already using a Maximum Wait Time here, so I'm not sure if Sample 09 has much value. They're almost the same.

Reply:
Unfortunately, it needs it because of the "Last" use.     We explain better in sample 9.   We'll have to revisit.   For now, it works and we don't want to call attention to the partition receiver.